lib/remap-lang/src/path.rs

```diff
@@ -0,0 +1,445 @@
+use crate::{parser::Parser, Error as E};
+use std::fmt;
+use std::str::FromStr;
+
+#[derive(thiserror::Error, Clone, Debug, PartialEq)]
+pub enum Error {
+    #[error("unable to create path from alternative string: {0}")]
+    Alternative(String),
+}
+
+/// Provide easy access to individual [`Segment`]s of a path.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Ord, PartialOrd)]
+pub struct Path {
```

Comment:
I do get confused with two different `Path` structs in the same project. It's hard to keep track of which instance is being referred to. Is there a different naming strategy we could use? Perhaps `PathSegments`?

Reply:
I'm probably going to merge the expression and the `Path`, since there isn't really any reason to keep both around.